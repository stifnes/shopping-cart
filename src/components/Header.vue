<template>
  <header>
    <h1>Shopping List</h1>
    <button type="Button" @click="deleteProducts()" class="button-del">
      Delete
    </button>
  </header>
</template>

<script>
export default {
  methods: {
    deleteProducts() {
      this.$emit("productDeleted");
    },
  },
};
</script>
<style lang="scss" scoped>
header {
  padding: 1rem;
  background-color: #fff;
  box-shadow: 0 5px 10px 0 rgba(0, 64, 128, 0.05%);
  display: flex;
  align-self: center;
  justify-content: space-between;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 65px;
  box-sizing: border-box;
  h1 {
    font-size: 22px;
    font-weight: 700;
    margin: 0;
  }
  button {
    background-color: rgb(139, 20, 60);
    border: none;
    border-radius: 8px;
    padding: 0 1rem;
    color: #fff;
  }
}
</style>
