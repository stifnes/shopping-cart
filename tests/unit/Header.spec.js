import { shallowMount } from "@vue/test-utils";
import Header from "@/components/Header.vue";

describe("Header", () => {

  let wrapper;
  
  beforeEach(() =>{
    wrapper = shallowMount(Header);
  });

  afterEach(() =>{
    wrapper.destroy();
  });

  it("should delete products on click", async() => {
    await wrapper.find('.button-del').trigger('click');
    expect(wrapper.vm.deleteProducts).$emit;
  })

  it("has the expected HTML structure", () =>{
    expect(wrapper.element).toMatchSnapshot();
  });
});
