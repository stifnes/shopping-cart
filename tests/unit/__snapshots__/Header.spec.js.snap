// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header has the expected HTML structure 1`] = `
<header>
  <h1>
    Shopping List
  </h1>
   
  <button
    class="button-del"
    type="Button"
  >
    
    Delete
  
  </button>
</header>
`;
